src/main/scala/io/druid/indexer/spark/SparkBatchIndexTask.scala

```diff
@@ -407,6 +414,46 @@ object SparkBatchIndexTask
         }
       )
 
+      log.info("Initializing emitter for metrics")
+
+      val service = task.getProperties.getProperty("com.metamx.emitter.service", "druid/batch/index")
+      val host = task.getProperties.getProperty("com.metamx.emitter.host", "127.0.0.1")
+
+      val emitter = new ServiceEmitter(service, host, createEmitter(task.getProperties, lifecycle))
```

Comment:
`EmittingLogger` can emit alerts, which is what makes it different than a normal logger.

The `ServiceEmitter` is made to do what you're doing here with it, and should probably have the injected one used.